File: dataLayer/entities/recognition.py
from sqlalchemy import Column, String, Integer, Date, ForeignKey
from sqlalchemy.orm import relationship

from dataLayer.database_connection import Base
from dataLayer.entities.neural_network import NeuralNetwork
from dataLayer.entities.recognition_image import RecognitionImage


class Recognition(Base):
    __tablename__ = 'Recognition'

    id = Column('Id', Integer, primary_key=True)
    name = Column('Name', String)
    neural_network_id = Column("NeuralNetworkId", Integer, ForeignKey("NeuralNetworks.Id"))
    neural_network = relationship(NeuralNetwork)
    statusId = Column('StatusId', Integer, ForeignKey("Status.Id"))
    images = relationship(RecognitionImage)

File: DataLayer/Entities/Recognition.cs
using System;
using System.Collections.Generic;
using DataLayer.Entities.Common;

namespace DataLayer.Entities
{
    public class Recognition : EntityBase
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public int? NeuralNetworkId { get; set; }
        public NeuralNetwork NeuralNetwork { get; set; }
        public int? StatusId { get; set; }
        public Status Status { get; set; }
        public DateTime? CompletionTime { get; set; }
        public IEnumerable<RecognitionImage> Images { get; set; }
        public IEnumerable<RecognitionResult> RecognitionResults { get; set; }
    }
}